stride/plotting/plot_traces.py

```diff
@@ -50,6 +50,113 @@ def plot_trace(*args, axis=None, colour='black', line_style='solid', title=None,
     return axis
 
 
+def plot_magnitude_spectrum(sampling_rate, *args, skip=1, time_range=None, norm=True,
+                            colour='black', line_style='solid', title=None, axis=None,
+                            per_trace=False, **kwargs):
+    """
+    Plot the magnitude spectrum of wavelet traces in dB scale.
+
+    Parameters
+    ----------
+    sampling_rate : float
+        Sampling rate of the wavelet signal in Hz.
+    args : arrays
+        Optional trace ID grid, frequency grid, and signal data.
+    skip : int, optional
+        Traces to skip, defaults to 1.
+    time_range : tuple, optional
+        Time range to plot, defaults to all.
+    norm : bool, optional
+        Normalize the whole gather, defaults to True.
+    colour : str, optional
+        Colour for the traces, defaults to black.
+    line_style : str, optional
+        Line style for the plot, defaults to solid.
+    title : str, optional
+        Plot title, defaults to None.
+    per_trace : bool, optional
+        If True, normalize each trace individually.
+    axis : matplotlib axis, optional
+        Axis on which to plot, defaults to a new figure.
+
+    Returns
+    -------
+    matplotlib axis
+        Axis with the plotted data.
+    """
+    try:
+        if not os.environ.get('DISPLAY', None):
+            raise ModuleNotFoundError
+        import matplotlib.pyplot as plt
+    except ModuleNotFoundError:
+        return None, None
+
+    # Set up axis
+    if axis is None:
+        figure, axis = plt.subplots(1, 1)
+
+    # Extract arguments
+    if len(args) > 2:
+        trace_ids, freq_grid, signal_data = args
+    elif len(args) > 1:
+        trace_ids = None
+        freq_grid, signal_data = args
+    else:
+        trace_ids = None
+        signal_data = args[0]
+
+    signal_data = signal_data[::skip]
+    if trace_ids is not None:
+        trace_ids = trace_ids[::skip]
+
+    # Apply time range if specified
+    if time_range:
+        start, end = time_range
+        start_idx, end_idx = int(start * sampling_rate), int(end * sampling_rate)
+        signal_data = signal_data[:, start_idx:end_idx]
+
+    # Normalize signal data if required
+    if norm:
+        signal_data = signal_data / np.max(np.abs(signal_data), axis=-1, keepdims=True)
```

Comment:
You need to apply two possible normalisations: by gather and by trace. See `plot_gather` for reference